show/main.py

```diff
@@ -1385,10 +1385,37 @@ def ports(portname, verbose):
 # 'bgp' subcommand ("show runningconfiguration bgp")
 @runningconfiguration.command()
 @click.option('--verbose', is_flag=True, help="Enable verbose output")
-def bgp(verbose):
+@click.option('--namespace', '-n', 'namespace', required=False, default=None, type=str, show_default=False, 
+              help='Option needed for multi-asic only - provide namespace name or all')
+def bgp(namespace, verbose):
     """Show BGP running configuration"""
-    cmd = 'sudo {} -c "show running-config"'.format(constants.RVTYSH_COMMAND)
-    run_command(cmd, display_cmd=verbose)
+    if multi_asic.is_multi_asic() and (namespace not in multi_asic.get_namespace_list() or namespace != 'all'):
+        ctx = click.get_current_context()
+        ctx.fail("-n/--namespace option required. provide namespace from list {}, or give '-n all'".format(multi_asic.get_namespace_list()))
```

Comment:
Would it be better to have default for multi-asic whereif the user do not specify anything such as "-n all" it defaults to display for all ASICs instead of forcing the user to have to specify "-n all"?
What does the other multi-asic show commands behave when it comes to BGP related show commands? Do they all forces the user to always have to specify -n option?  We should keep similar default behavior...

Reply:
@gechiang your point sounds fair. Other cmds can run without 'namespace' specified and show all.
But here is the blocker:
- other cmd like `show ip interfaces` uses `multi_asic_click_options` that can take optional namespaces, while this also gives a 'display' option that rvtysh cmd cannot take for `show run bgp`
- even with `required = false` inside option `-n`, it won't work as an optional option, it still requires '-n' to be given for multi-asic as a must have.

will try to see how to make `-n` optional.

Reply:
This issue is addressed.
`show run bgp` by default will print for all asics.
`show run bgp -n asicx` will print for asicx